clippy_lints/src/manual_let_else.rs

```diff
@@ -185,18 +185,7 @@ fn expr_diverges(cx: &LateContext<'_>, expr: &'_ Expr<'_>) -> bool {
 /// Returns true if the two spans come from different macro sites,
 /// or one comes from an invocation and the other is not from a macro at all.
 fn from_different_macros(span_a: Span, span_b: Span) -> bool {
-    // This pre-check is a speed up so that we don't build outer_expn_data unless needed.
-    match (span_a.from_expansion(), span_b.from_expansion()) {
-        (false, false) => return false,
-        (true, false) | (false, true) => return true,
-        // We need to determine if both are from the same macro
-        (true, true) => (),
-    }
-    let data_for_comparison = |sp: Span| {
-        let expn_data = sp.ctxt().outer_expn_data();
-        (expn_data.kind, expn_data.call_site)
-    };
-    data_for_comparison(span_a) != data_for_comparison(span_b)
+    span_a.ctxt() != span_b.ctxt()
```

Comment:
Do you really need a function for this? I think you can just inline it.

Reply:
I think it's still good to have a function because then it's clear what the code is doing. But I will inline it...